web/src/components/TaskHeader.test.tsx

```diff
@@ -192,6 +193,456 @@ describe("<TaskHeader />", () => {
     });
   });
 
+  describe("tax registration modal", () => {
+    it("opens when tax task is marked complete", () => {
+      const taxTask = generateTask({ id: taxTaskId, stepNumber: 1 });
+
+      const userData = generateUserData({
+        taskProgress: {
+          [taxTaskId]: "NOT_STARTED",
+        },
+        preferences: generatePreferences({ roadmapOpenSections: ["START"] }),
+      });
+
+      useMockRoadmap({
+        steps: [generateStep({ stepNumber: 1, section: "START" })],
+        tasks: [taxTask],
+      });
+
+      renderTaskHeader(taxTask, userData);
+      selectCompleted();
+
+      expect(screen.getByText(Config.taxRegistrationModal.title)).toBeInTheDocument();
+    });
+
+    it("doesn't update the task status when closed", async () => {
+      const taxTask = generateTask({ id: taxTaskId, stepNumber: 1 });
+
+      const userData = generateUserData({
+        taskProgress: {
+          [taxTaskId]: "NOT_STARTED",
+        },
+        preferences: generatePreferences({ roadmapOpenSections: ["PLAN"] }),
+      });
+
+      useMockRoadmap({
+        steps: [generateStep({ stepNumber: 1, section: "PLAN" })],
+        tasks: [taxTask],
+      });
+
+      renderTaskHeader(taxTask, userData);
+      selectCompleted();
+
+      fireEvent.click(screen.getByTestId("modal-button-secondary"));
```

Comment:
it's better to select by text (in this case, `taxRegistrationModal.cancelButtonText`) when possible